lte/cloud/go/services/subscriberdb/storage/subscribers.go

```diff
@@ -0,0 +1,302 @@
+		_, err := l.builder.CreateTable(subscriberTableName).
+			IfNotExists().
+			Column(subscriberNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subscriberNidCol, subscriberSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store table")
+		}
+		_, err = l.builder.CreateTable(subscriberTmpTableName).
+			IfNotExists().
+			Column(subscriberNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subscriberNidCol, subscriberSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store tmp table")
+		}
+		return nil, nil
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// InitiateUpdate prepares the db tables for a batch update.
+func (l *SubStore) InitiateUpdate() error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.
+			Delete(subscriberTmpTableName).
+			RunWith(tx).
+			Exec()
+		return nil, errors.Wrapf(err, "clear sub protos tmp table")
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// InsertMany inserts a batch of subscriber data into the temporary table.
+func (l *SubStore) InsertMany(network string, subProtos []*lte_protos.SubscriberData) error {
+	// NOTE: Caller of this function should guard against the case where
+	// len(subProtos) > maxEntitiesLoadSize of configurator
+	if len(subProtos) == 0 {
+		return nil
+	}
+	insertQuery := l.builder.
+		Insert(subscriberTmpTableName).
+		Columns(subscriberNidCol, subscriberSidCol, subscriberProtoCol)
+	errs := &multierror.Error{}
+	for _, subProto := range subProtos {
+		marshaled, err := proto.Marshal(subProto)
+		if err != nil {
+			multierror.Append(errs, errors.Wrapf(err, "serialize subproto of ID %+v", lte_protos.SidString(subProto.Sid)))
+			continue
+		}
+		insertQuery = insertQuery.Values(network, lte_protos.SidString(subProto.Sid), marshaled)
+	}
+	if errs.ErrorOrNil() != nil {
+		return errs
+	}
+
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := insertQuery.RunWith(tx).Exec()
+		return nil, errors.Wrapf(err, "insert sub protos into store for network %+v", network)
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// ApplyUpdate applies all subscriber data updates onto the db table and completes the batch update.
+func (l *SubStore) ApplyUpdate(network string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		// HACK: hard coding part of this sql query because there currently doesn't exist good support
+		// for "WHERE (row NOT IN other_table)" with squirrel
+		_, err := l.builder.
```

Comment:
Similar to a previous comment: add a comment that gives a summary of the SQL that this builder actually generates, for easy skimming